### ISSUE TITLE
Parsing bad analysis_options.yaml should not stop generating the documentation.

### ISSUE THREAD

**Issue by isoos** [dart-lang/dartdoc#2241]
One particular report is a duplicate key in `analysis_options.yaml`:
https://github.com/dart-lang/pub-dev/issues/3708

**Comment by isoos** [comment]
A similar issue with another package:

````
"Error while running dartdoc for imgur 1.1.1.
exitCode: 255
stdout: 
stderr: Unhandled exception:
Error on line 13, column 13: Undefined alias.
   ╷
13 │   exclude: [*.g.dart]
   │             ^^^^^^^^
   ╵
#0      Loader._loadAlias (package:yaml/src/loader.dart:108:5)
#1      Loader._loadNode (package:yaml/src/loader.dart:80:16)
#2      Loader._loadSequence (package:yaml/src/loader.dart:140:20)
#3      Loader._loadNode (package:yaml/src/loader.dart:84:16)
#4      Loader._loadMapping (package:yaml/src/loader.dart:163:19)
#5      Loader._loadNode (package:yaml/src/loader.dart:86:16)
#6      Loader._loadMapping (package:yaml/src/loader.dart:163:19)
#7      Loader._loadNode (package:yaml/src/loader.dart:86:16)
#8      Loader._loadDocument (package:yaml/src/loader.dart:62:20)
#9      Loader.load (package:yaml/src/loader.dart:54:20)
#10     loadYamlDocument (package:yaml/yaml.dart:51:25)
#11     loadYamlNode (package:yaml/yaml.dart:42:5)
#12     loadYaml (package:yaml/yaml.dart:34:5)
#13     new _FilePackageMeta (package:dartdoc/src/package_meta.dart:295:26)
#14     PubPackageMeta.fromDir (package:dartdoc/src/package_meta.dart:223:27)
````

**Comment by jcollins-g** [comment]
dartdoc should at most generate an error here (possibly a warning if non-fatal) rather than blowing out with an exception.

**Comment by isoos** [comment]
I think this has been fixed in dartdoc 0.32.4 or earlier.
